### PR TITLE
[HOPSWORKS-2195] Attach Jupyter configuration to notebook as Xattr

### PR DESCRIPTION
**Pull request by gibchikafa**

### CODE REVIEW

File: sparkmagic/sparkmagic/livyclientlib/livysession.py [hunk 1]
```diff
@@ -335,3 +335,21 @@ def get_html_link(text, url):
...
+                       "project" + hopsconstants.DELIMITERS.SLASH_DELIMITER + os.environ["HOPSWORKS_PROJECT_ID"] + \
+                       hopsconstants.DELIMITERS.SLASH_DELIMITER + "jupyter/attachConfiguration" + \
+                       hopsconstants.DELIMITERS.SLASH_DELIMITER + os.environ["HADOOP_USER_NAME"] + \
+                       hopsconstants.DELIMITERS.SLASH_DELIMITER + kernel_id
+        self.ipython_display.writeln(resource_url)
+        self.ipython_display.writeln("Repeat" + resource_url)
+        connection = self.hopsworks_rest_connector.get_http_connection(https=True)
+        headers = {}
+        jwt_text = self.hopsworks_rest_connector.get_jwt()
+        print(jwt_text)
```
Comment (tkakantousis) [inline sparkmagic/sparkmagic/livyclientlib/livysession.py]:
```suggestion
```

File: sparkmagic/sparkmagic/livyclientlib/livysession.py [hunk 2]
```diff
@@ -335,3 +335,21 @@ def get_html_link(text, url):
...
+
+    def attach_jupyter_configuration_to_notebook(self, kernel_id):
+        method = hopsconstants.HTTP_CONFIG.HTTP_PUT
+        resource_url = hopsconstants.DELIMITERS.SLASH_DELIMITER + \
+                       hopsconstants.REST_CONFIG.HOPSWORKS_REST_RESOURCE + hopsconstants.DELIMITERS.SLASH_DELIMITER + \
+                       "project" + hopsconstants.DELIMITERS.SLASH_DELIMITER + os.environ["HOPSWORKS_PROJECT_ID"] + \
+                       hopsconstants.DELIMITERS.SLASH_DELIMITER + "jupyter/attachConfiguration" + \
+                       hopsconstants.DELIMITERS.SLASH_DELIMITER + os.environ["HADOOP_USER_NAME"] + \
+                       hopsconstants.DELIMITERS.SLASH_DELIMITER + kernel_id
+        self.ipython_display.writeln(resource_url)
```
Comment (robzor92) [inline sparkmagic/sparkmagic/livyclientlib/livysession.py]: Do we need to print this in the notebook?

File: sparkmagic/sparkmagic/livyclientlib/livysession.py [hunk 3]
```diff
@@ -335,3 +335,21 @@ def get_html_link(text, url):
...
+        self.ipython_display.writeln("Repeat" + resource_url)
```
Comment (robzor92) [inline sparkmagic/sparkmagic/livyclientlib/livysession.py]: And print this?

**Review by robzor92 — CHANGES_REQUESTED** [review]

File: sparkmagic/sparkmagic/livyclientlib/livysession.py (hunk 3 above)
Comment (gibchikafa) [reply sparkmagic/sparkmagic/livyclientlib/livysession.py]: It was for debugging. I will remove it.

**Review by robzor92 — APPROVED** [review]

**Pull request merged**